mobile/src/main/java/org/openhab/habdroid/ui/ViewExtensions.kt

```diff
@@ -120,14 +121,19 @@ fun RemoteViews.duplicate(): RemoteViews {
 }
 
 fun MaterialButton.setTextAndIcon(connection: Connection, mapping: LabeledValue) {
-    contentDescription = mapping.label
-    val iconUrl = mapping.icon?.toUrl(context, true)
+    setTextAndIcon(connection, mapping.label, mapping.icon)
+}
+
+fun MaterialButton.setTextAndIcon(connection: Connection, label: String, icon: IconResource?) {
+    contentDescription = label
+    val iconUrl = icon?.toUrl(context, true)
     if (iconUrl == null) {
-        icon = null
-        text = mapping.label
+        this.icon = null
+        text = label
         return
     }
     val iconSize = context.resources.getDimensionPixelSize(R.dimen.section_switch_icon)
+    val thisButton = this
```

Comment:
We usually do `this@MaterialButton` instead of using a variable.

Reply:
Kotlin noob here! Thanks for the tip!